inference-engine/thirdparty/clDNN/src/gpu/loop_gpu.cpp

```diff
@@ -0,0 +1,453 @@
+                        CLDNN_ERROR_MESSAGE(instance.id(), "incompatible datatypes");
+                    }
+                    body_network->set_input_data(input_pm.internal_id, memory);
+                }
+
+                // checking if memory is a destination of a backedge
+                const auto& back_edges = node.get_back_edges();
+                for (const auto& back_edge : back_edges) {
+                    if (input_pm.internal_id != back_edge.to) {
+                        continue;
+                    }
+                    //find corresponding input of the backedge
+                    for (const auto& body_output : body_network->get_outputs()) {
+                        if (body_output->id() != back_edge.from) {
+                            continue;
+                        }
+                        const int max_iteration = node.get_max_iteration();
+                        auto from_mems = find_cropped_mem(back_edge.from, instance);
+                        if (from_mems.empty()) { // backedge output which does not need concatenation
+                            // input memory = output memory = loop output memory
+                            memory_impl::ptr loop_input_mem = get_outer_output_memory(instance, input_pm.external_id);
+                            for (auto& output : body_network->get_outputs()) {
+                                if (output->id() != back_edge.from) {
+                                    continue;
+                                }
+                                const auto output_primitive_mapping = find_primitive_mappings(back_edge.from);
+                                if (output_primitive_mapping.empty()) {
+                                    continue;
+                                }
+                                memory_impl::ptr loop_output_mem = get_outer_output_memory(instance, output_primitive_mapping.front().external_id);
+                                body_network->set_input_data(back_edge.to, *loop_output_mem);
+                                body_network->set_output_memory(back_edge.from, *loop_output_mem);
+                                copy_buffer(*loop_input_mem, *loop_output_mem, loop_input_mem->get_layout().count());
+                                break;
+                            }
+
+                        } else {
+                            memory_impl::ptr initial_mem = get_outer_output_memory(instance, input_pm.external_id);
+                            backedge_mem.emplace_back(body_output, initial_mem);
+                            for (int i = 0 ; i < max_iteration; ++i) {
+                                memory_impl::ptr from_mem = from_mems.at(i);
+                                backedge_mem.back().add_backedge_from_mem(from_mem);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // extract int from data primitive
```

Comment:
I think "int" is too naive.. how about get_scalar_value?

Also why the varialbe name is "trip_count" thought it is being used for other purpose?

Reply:
will be renamed to get_scalar_value

the variable name "trip_count" is from [IR loop operation spec](https://docs.openvinotoolkit.org/latest/openvino_docs_ops_infrastructure_Loop_5.html)

Reply:
My question was why it was trip_count though it was bing used for other data "too". Seems that it it is outdated. closing